configure.ac

```diff
@@ -110,57 +94,188 @@ SET_ARCH([$build_cpu],[build_arch])
+                              [do not provide the weak 'backtrace' symbol
+                               @<:@default=no@:>@])],
+              [],
+              [enable_weak_backtrace=yes]
 )
+AC_MSG_RESULT([$enable_weak_backtrace])
+AM_CONDITIONAL([CONFIG_WEAK_BACKTRACE], [test "x$enable_weak_backtrace" = xyes])
+AM_COND_IF([CONFIG_WEAK_BACKTRACE], [
+  AC_DEFINE([CONFIG_WEAK_BACKTRACE], [1], [Define if the weak 'backtrace' symbol is provided.])
+])
 
-AC_MSG_CHECKING([if we should build libunwind-ptrace])
-AC_MSG_RESULT([$enable_ptrace])
 
-AC_ARG_ENABLE(setjmp,
-	AS_HELP_STRING([--enable-setjmp],[building libunwind-setjmp library]),,
-        [AS_IF([test x$target_arch = x$host_arch], [enable_setjmp=yes], [enable_setjmp=no])]
+AC_MSG_CHECKING([if unwind.h should be exported])
+AC_ARG_ENABLE([unwind-header],
+              [AS_HELP_STRING([--disable-unwind-header],
+                              [do not export the 'unwind.h' header
+                               @<:@default=no@:>@])],
+              [],
+              [enable_unwind_header=yes]
 )
+AC_MSG_RESULT([$enable_unwind_header])
+AM_CONDITIONAL(BUILD_UNWIND_HEADER, test "x$enable_unwind_header" = xyes)
 
-AC_ARG_ENABLE(tests,
- AS_HELP_STRING([--disable-tests],[Disable tests build]),,
- [enable_tests=yes])
+AC_MSG_CHECKING([whether to support UNW_CACHE_PER_THREAD])
+AC_ARG_ENABLE([per-thread-cache],
+              [AS_HELP_STRING([--enable-per-thread-cache],
+                              [build with support for UNW_CACHE_PER_THREAD
+                               (which imposes a high TLS memory usage)
+                               @<:@default=no@:>@])],
+              [],
+              [enable_per_thread_cache=no]
+)
+AC_MSG_RESULT([$enable_per_thread_cache])
+AS_IF([test x$enable_per_thread_cache = xyes],
+      [AC_DEFINE([HAVE___CACHE_PER_THREAD], 1, [Define to 1 if --enable-per-thread-cache])]
+)
 
-AC_ARG_ENABLE(weak-backtrace,
- AS_HELP_STRING([--disable-weak-backtrace],[Do not provide the weak 'backtrace' symbol.]),,
- [enable_weak_backtrace=yes])
+AC_MSG_CHECKING([if testsuite should be built])
+AC_ARG_ENABLE([tests],
+              [AS_HELP_STRING([--disable-tests],
+                              [disable building tests @<:@default=no@:>@])],
+              [],
+              [enable_tests=yes]
+)
+AC_MSG_RESULT([$enable_tests])
+AM_CONDITIONAL([CONFIG_TESTS], [test x$enable_tests = xyes])
+AM_COND_IF([CONFIG_TESTS], [
+  old_LIBS="$LIBS"
+  AC_MSG_NOTICE([--- Checking for extra libraries linked to tests ---])
+  AC_SEARCH_LIBS([dlopen], [dl],
+                 [AS_IF([test "$ac_cv_search_dlopen" != "none required"],
+                        [AC_SUBST([DLLIB], ["$ac_cv_search_dlopen"])])])
+  AC_SEARCH_LIBS([pthread_key_create], [pthread],
+                 [AS_IF([test "$ac_cv_search_pthread_key_create" != "none required"],
+                        [AC_SUBST([PTHREADS_LIB], ["$ac_cv_search_pthread_key_create"])])])
```

Comment:
why not using https://www.gnu.org/software/autoconf-archive/ax_pthread.html?

Reply:
ax_pthread.m4 is almost larger than configure.ac. This project currently has no infrastructure in place to handle third-party autoconf macro files (not that that is hard to set up) and the checks in ax_pthread.m4 seem like overkill for the purpose here, which is a few unit tests. Its primary purpose if to handle obscure edge cases on operating systems we don't even support, like AIX and Digital Unix. 

It's a good suggestion worth considering, but on the whole I think a simple test for whether `-lpthread` is necessary or not will improve portability while minimally impacting maintainability and build times.